images/inference-services/seldon-core/run.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
+set -e
+set -u
+set -o pipefail
+
+# load org and project from repository if exists,
+# if not, set them as a random string
+if [ -e ".fuseml/_project" ]; then
+    export ORG=$(cat .fuseml/_org)
+    export PROJECT=$(cat .fuseml/_project)
+else
+    export ORG=$(tr -dc a-z0-9 </dev/urandom | head -c 6 ; echo '')
+    export PROJECT=$(tr -dc a-z0-9 </dev/urandom | head -c 6 ; echo '')
+fi
+
+export S3_ENDPOINT=${MLFLOW_S3_ENDPOINT_URL/*:\/\//}
+
+mc alias set minio ${MLFLOW_S3_ENDPOINT_URL} ${AWS_ACCESS_KEY_ID} ${AWS_SECRET_ACCESS_KEY}
+model_bucket="minio${FUSEML_MODEL//s3:\//}"
+
+export PROTOCOL_VERSION="v1"
+export PREDICTOR=${FUSEML_PREDICTOR}
+if [ "${PREDICTOR}" = "auto" ]; then
+    if ! mc stat ${model_bucket}/MLmodel &> /dev/null ; then
+        echo "No MLmodel found, cannot auto detect predictor"
+        exit 1
+    fi
+
+    PREDICTOR=$(mc cat ${model_bucket}/MLmodel | awk -F '.' '/loader_module:/ {print $2}')
+fi
+
+sd="${ORG}-${PROJECT}"
+case $PREDICTOR in
+    sklearn)
+        if ! mc ls ${model_bucket} | grep -q "model.joblib"; then
+            mc cp ${model_bucket}/model.pkl ${model_bucket}/model.joblib
+        fi
+        export PREDICTOR_SERVER="SKLEARN_SERVER"
+        prediction_url_path="${sd}/api/v1.0/predictions"
+        ;;
+esac
+
+#TODO parametrize prediction method
+
+# Gateway has host info in the form of '*.seldon.172.18.0.2.nip.io' so we need to add a prefix
+domain=$(kubectl get Gateway seldon-gateway -n ${FUSEML_ENV_WORKFLOW_NAMESPACE} -o jsonpath='{.spec.servers[0].hosts[0]}')
+export ISTIO_HOST="${ORG}.${PROJECT}${domain/\*/}"
+
+envsubst < /root/template.sh | kubectl apply -f -
+
+# next wait commands sometimes fail when the resource seems to be missing
+sleep 1
```

Comment:
Instead of this sleep I suggest trying to get the resource until it succeeds, waiting for one second might not be enough on some cases.

Reply:
I am actually trying that, that's the next command `kubectl rollout` ... works most of the time but I've still seen it failed once.